### PR TITLE
[100] Create infrastructure for saving files with Azure Blob, [110] Add the lot to the watch list (Backend), [112] View lot profile (Backend)

### PR DESCRIPTION
**Pull request by khasanrashidov**
### Where

[NL-100 Create infrastructure for saving files with Azure Blob](https://jjbcapital.atlassian.net/browse/NL-100)
[NL-110 Add the lot to the watch list (Backend)](https://jjbcapital.atlassian.net/browse/NL-110)
[NL-112 View lot profile (Backend)](https://jjbcapital.atlassian.net/browse/NL-112)
### What

Set up the Azure Blob Storage infrastructure for saving photos and additional documents.

### How

Implemented Azure Blob Storage infrastructure for saving photos and additional documents. Every get, update, create, delete 
lot endpoint has blob integrated with it.

### Affected Area

Auctionify.API.Controllers
Auctionify.Application.Common
Auctionify.Application.Features
Auctionify.Infrastructure.Services
Auctionify.Infrastructure.Repositories

### PR Checklist

_Build_

- [x] Build Succeeded

### CODE REVIEW

File: src/Auctionify.API/appsettings.Development.json [hunk 1]
```diff
@@ -8,6 +8,12 @@
   "ConnectionStrings": {
     "DefaultConnection": "Server=(localdb)\\mssqllocaldb;Database=Auctionify;Trusted_Connection=True;MultipleActiveResultSets=true"
   },
+  "AzureBlobStorageSettings": {
+    "ConnectionString": "DefaultEndpointsProtocol=https;AccountName=auctionify;AccountKey=swVedDpxhGOTiDo6fdHNpTqrdt5LF2lPmKeVFT3N3LjgmDqKEpic/doIWMPFr87xAb30EggwNFqN+AStmgLbjw==;EndpointSuffix=core.windows.net",
```
Comment (SevenDart) [inline src/Auctionify.API/appsettings.Development.json:12]: You mustn't provide any credentials in public repositories, it is a security leak. You must drop this commit from the repository to remove all mentions about these credentials. When push such things as ConnectionString, use empty or default values that are not actual credentials. Credentials and app keys are secrets that should be managed separately. There a few different approaches to manage them. There are some links to these approaches in documentation about configuration: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-7.0#security-and-user-secrets

**Review by SevenDart — CHANGES_REQUESTED** [review]

File: src/Auctionify.API/appsettings.Development.json (hunk 1 above)
Comment (khasanrashidov) [reply src/Auctionify.API/appsettings.Development.json:12]: yes, I know, that was accidentally, I didn't notice it).. no worries, I rotated the key 🗝

**Comment by khasanrashidov** [comment]
Fixed, corrected, refactored as required

**Review by SevenDart — APPROVED** [review]

File: src/Auctionify.Application/Features/Lots/Commands/DeleteLotFile/DeleteLotFileCommand.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+							+ 1,
+						lastSlashIndex
+							- (
+								photoUrl.IndexOf(_azureBlobStorageOptions.ContainerName)
+								+ _azureBlobStorageOptions.ContainerName.Length
+								+ 1
+							)
+					);
+
+					Console.WriteLine(filePath);
```
Comment (SevenDart) [inline src/Auctionify.Application/Features/Lots/Commands/DeleteLotFile/DeleteLotFileCommand.cs:77]: Remove Console.WriteLine

**Review by SevenDart — CHANGES_REQUESTED** [review]

File: src/Auctionify.Application/Features/Lots/Commands/DeleteLotFile/DeleteLotFileCommand.cs (hunk 2 above)
Comment (khasanrashidov) [reply src/Auctionify.Application/Features/Lots/Commands/DeleteLotFile/DeleteLotFileCommand.cs:77]: fixed, removed

File: src/Auctionify.Application/Common/Interfaces/IEmailService.cs [hunk 3]
```diff
@@ -2,6 +2,6 @@
 {
 	public interface IEmailService
 	{
-		Task SendEmailAsync(string toEmail, string subject, string content);
+		public Task SendEmailAsync(string toEmail, string subject, string content);
```
Comment (SevenDart) [inline src/Auctionify.Application/Common/Interfaces/IEmailService.cs:5]: There is no any sense in adding keyword "public" when defining interface methods, all members are public by default. It has sense only if you want to use another modifier for some methods, to distinct public from private/protected.

File: src/Auctionify.Application/Common/Interfaces/IWatchlistService.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Auctionify.Application.Common.Interfaces
+{
+	public interface IWatchlistService
+	{
+		public Task<bool> IsLotInUserWatchlist(int lotId, int userId, CancellationToken cancellationToken);
```
Comment (SevenDart) [inline src/Auctionify.Application/Common/Interfaces/IWatchlistService.cs:5]: Same with previous interface

**Review by SevenDart — APPROVED** [review]

File: src/Auctionify.Application/Common/Interfaces/IEmailService.cs (hunk 3 above)
Comment (khasanrashidov) [reply src/Auctionify.Application/Common/Interfaces/IEmailService.cs:5]: fixed

File: src/Auctionify.Application/Common/Interfaces/IWatchlistService.cs (hunk 4 above)
Comment (khasanrashidov) [reply src/Auctionify.Application/Common/Interfaces/IWatchlistService.cs]: fixed

**Review by SevenDart — DISMISSED** [review]

**Review by SaidganiUsm — APPROVED** [review]

**Pull request merged**